velox/vector/BaseVector.cpp

```diff
@@ -1007,6 +1031,98 @@ std::string printIndices(
   return out.str();
 }
 
+// static
+void BaseVector::transposeIndices(
+    const vector_size_t* base,
+    vector_size_t size,
+    const vector_size_t* indices,
+    vector_size_t* result) {
+  constexpr int32_t kBatch = xsimd::batch<int32_t>::size;
+  int32_t i = 0;
+  for (; i + kBatch <= size; i += kBatch) {
+    auto indexBatch = xsimd::load_unaligned(indices + i);
+    simd::gather(base, indexBatch).store_unaligned(result + i);
+  }
+  if (i < size) {
+    auto indexBatch = xsimd::load_unaligned(indices + i);
+    auto mask = simd::leadingMask<int32_t>(size - i);
+    simd::maskGather(
+        xsimd::batch<int32_t>::broadcast(0), mask, base, indexBatch)
```

Comment:
this won't overwrite beyond the size? Or the buffer is padding to xsimd batch aligned? thanks!